### DISCUSSION TITLE
Relative Time and Indexes

### DISCUSSION THREAD

**Discussion by TheMadKow** [ravendb/ravendb#13712] [Q&A]
Hi,
I know I can't use time compare with index, but I wonder if there is an easy solution for my current question.

Say we have a todo system, and I've created some todos with a due date for a person (#5).
I can query like this - "where todo.dueDate < now" to know that this person haven't completed his or hers task.

I need this data in a lot of places, and It probably won't change that fast, but the compare to NOW will force the query to re-run.

So what I want is an "Index" that says "Person ID #5 has 2 overdue todos, 3 open todos ..."

My current instinct is to create some service worker that fires up on the next due date and update some "snapshot" object for the person. It'll actually call for said queries and update the snapshot object every once in a while.

The main reason is performance, I don't want to query with "now" every time I need this data. If I create this snapshot I can utilize aggressive caching, and thus, won't contact the DB at all, until next due date.

At "0 cost" I'll know this persons status.

So the question is, Can I utilize the indexes in some other way to help me out ?

Thanks,
Eliran.

**Comment by ayende** [comment]
You can use a facet query, like so:

```
from index 'ToDoTasks'
select facet(DueDate <= $now, DueDate > $now)
```

That will give you both numbers in one shot easily.

**Reply by TheMadKow** [reply]
Thanks, I'll delve deeper into facets, thanks ! :)
